contracts/NftToken.sol

```diff
@@ -17,47 +17,92 @@ contract Creature is ERC721("My NFT Collection", "MNFTC"), Ownable {
     event Buy(address indexed to, uint256 tokenId, uint256 cost);
     event Withdraw(address indexed to, uint256 amount);
     
-    
-    constructor(string memory initBaseURI) {
+    /**
+     * @dev Initializes the base link and initial cost
+     *
+     * @param initBaseURI Base URI
+     * @param initCost Initial cost
+     */
+    constructor(string memory initBaseURI, uint256 initCost) {
         setBaseURI(initBaseURI);
```

Comment:
It's not good to have setter for setting a base URI.
Change to internal function and allow it to be called in contract construction only